src/sql/workbench/api/browser/mainThreadConnectionManagement.ts

```diff
@@ -182,6 +182,38 @@ export class MainThreadConnectionManagement extends Disposable implements MainTh
 		return connection;
 	}
 
+	public $openChangePasswordDialog(initialConnectionProfile: IConnectionProfile): void {
+		// Need to have access to getOptionsKey, so recreate profile from details.
+		let profile = new ConnectionProfile(this._capabilitiesService, initialConnectionProfile);
+		this._connectionManagementService.launchChangePasswordDialog(profile);
+	}
+
+	public $getConnectionProfileFromError(): Thenable<azdata.connection.ConnectionProfile> {
+		let profile = this._connectionManagementService.getConnectionProfileFromError();
+		if (!profile) {
+			return Promise.resolve(undefined);
+		}
+
+		// Need to convert profile into azdata friendly format
+		let connection: azdata.connection.ConnectionProfile = {
```

Comment:
Can you use `this.convertToConnectionProfile(profile)` instead? Also change that in `$getConnection()` to avoid maintaining multiple versions of this conversion.

Reply:
the convert function can be modified to save profile